### DISCUSSION TITLE
nn.GCNConv adjacency matrix: SparseTensor vs COO format tensor

### DISCUSSION THREAD

**Discussion by dxyzx0** [pyg-team/pytorch_geometric#3034] [Q&A]
I just find that in `GCNConv`, I can use the `COO` format to represent the adjacency matrix with `edge_index` and `edge_weight` or I can do it with sparse tensor `edge_index` and `edge_weight=None`.  I run into `RuntimeError: CUDA out of memory` even with `batch_size=1` since my graph has up to 300000 nodes. I wonder whether the latter input will consume less cuda memory than the former one?
![image](https://user-images.githubusercontent.com/28868457/130892326-cdfdef27-98e2-4e6e-8b3b-d50075599686.png)

**Comment by rusty1s** [comment]
Yes, the latter one will consume less memory on `GCNConv`. You can reduce the amount of memory further by performing adjacency normalization in a pre-processing step (e.g., via `torch_geometric.transforms.GCNNorm`), and initialize your GCN layers with `normalize=False`.

**Reply by dxyzx0** [reply]
But I must use `torch_sparse.SparseTensor` and CANNOT use `torch.sparse_coo_tensor` right?
However, If I use `torch_sparse.SparseTensor`, code won't be able to use `torch.cuda.amp` since `SparseTensor` won't be autocasted.

**Reply by rusty1s** [reply]
Let's move the discussion to here: https://github.com/rusty1s/pytorch_geometric/discussions/3036
